src/util/isoDateTime/isoDateTime.ts

```diff
@@ -0,0 +1,15 @@
+const getLocalizedYear = (date: Date, locale: string): string => {
+  const formatter = new Intl.DateTimeFormat(locale, { year: 'numeric' })
+  return formatter.format(date)
+}
+
+export const isISODateTime = (input: string): boolean => {
+  const isoDateTimeRegex: RegExp =
+    /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})(\.\d{1,6})?(([+-])(\d{2}):(\d{2})|Z)?$/
+
+  return isoDateTimeRegex.test(input) && new Date(input).toString() !== 'Invalid Date'
+}
+
+export const convertToLocalizedYearIfISODateTime = (input: string, locale: string): string => {
+  return isISODateTime(input) ? getLocalizedYear(new Date(input), locale) : input
+}
```

Comment:
You should not return the `input`: what about if I give `foo` as the `input` parameter of the `convertToLocalizedYearIfISODateTime` method? => the view will display `foo` as a temporal coverage, that is a wrong interpretation.
In a functional way, we would have used a `Either` task to capture both values (result, error) and then handle it in the view.
For now, I'll suggest either to throw an error or return a `Invalid date` that will be directly handle as a fallback in the view. (content must be validated with the product team)

Reply:
And please, keep ES6 syntax 😉

Reply:
@0xLodz The point is: _never trust the input_. The function `convertToLocalizedYearIfISODateTime` should function consistently in _any_ scenario, with the following expected behavior:

- It should return the year of the input date if it can be successfully parsed
- If it cannot be parsed, an error should be returned. The approach to return the error is at your discretion: _functional style_ (the prefered one) using an [Either<>](https://gcanti.github.io/fp-ts/modules/Either.ts.html) value, or _imperative style_ by throwing an error, as suggested by @fredericvilcot